src/nvim/os/time.c

```diff
@@ -144,6 +158,39 @@ struct tm *os_localtime(struct tm *result) FUNC_ATTR_NONNULL_ALL
   return os_localtime_r(&rawtime, result);
 }
 
+/// Portable version of POSIX ctime_r()
+///
+/// @param clock[in]
+/// @param result[out] Pointer to a 'char' where the result should be placed
+/// @param result_len length of result buffer
+/// @return human-readable string of current local time
+char *os_ctime_r(const time_t *restrict clock, char *restrict result,
+                 size_t result_len)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_NONNULL_RET
+{
+  struct tm clock_local;
+  struct tm *clock_local_ptr = os_localtime_r(clock, &clock_local);
+  // MSVC returns NULL for an invalid value of seconds.
+  if (clock_local_ptr == NULL) {
+    snprintf(result, result_len, "%s\n", _("(Invalid)"));
+  } else {
+    strftime(result, result_len, "%a %b %d %H:%M:%S %Y\n", clock_local_ptr);
```

Comment:
`ctime` always ends with a newline so I kept that behavior here. Vim's `get_ctime` has `add_newline` param to make this behavior optional.